cibyl/orchestrator.py

```diff
@@ -102,9 +96,76 @@ def select_source_method(self, system, argument):
         return Source.get_source_method(system.name.value, system_sources,
                                         argument.func)
 
+    def _consistent_environment(self, env):
+        """Check if an environment should be used according to user input.
+
+        :param env: Model to validate
+        :type env: :class:`.Environment`
+        :returns: Whether the environment is consistent with user input
+        :rtype: bool
+        """
+        user_env = self.parser.ci_args.get("env_name")
+        if user_env:
+            return env.name.value in user_env.value
+        return True
+
+    def _consistent_system(self, system):
+        """Check if a system should be used according to user input.
+
+        :param system: Model to validate
+        :type system: :class:`.System`
+        :returns: Whether the system is consistent with user input
+        :rtype: bool
+        """
+        is_valid_system = True
+        name = system.name.value
+
+        user_system_type = self.parser.ci_args.get("system_type")
+        if user_system_type:
```

Comment:
combine if statements 
Ex.
```
if user_system_type and system.system_type.value not in user_system_type.value:
```

Reply:
`_consistant_environment` and `_consistant_system` is part of validation  can we move it to somewhere else ? may be in `parser.py`?

And validate inside `Orchistrator init`.

Reply:
Maybe we could move it into a separate class, I'll look into it